### PR TITLE
enable password configuration via env variable

### PR DESCRIPTION
**Pull request by t-8ch**
This removes the possibility of leaking the password via the process table on
CI systems.

### CODE REVIEW

File: src/main/java/org/sonar/plugins/stash/StashRequestFacade.java [hunk 1]
```diff
@@ -185,8 +185,18 @@ public void postCommentPerIssue(String project, String repository, String pullRe
     }
   }
   
-  public StashCredentials getCredentials(){
-    return new StashCredentials(config.getStashLogin(), config.getStashPassword());
+  public StashCredentials getCredentials() throws StashConfigurationException {
```
Comment (mathieusimon) [inline src/main/java/org/sonar/plugins/stash/StashRequestFacade.java]: Should we not catch the exception in the same method?
For readability, I would tend to prefer full name variables like passwordEnv

Comment (t-8ch) [inline src/main/java/org/sonar/plugins/stash/StashRequestFacade.java]: I wanted to have the user see if they asked for the environment variable usage and then fail to provide it.

Comment (mathieusimon) [inline src/main/java/org/sonar/plugins/stash/StashRequestFacade.java]: What about catch the exception and logged a warning for the user then?

Comment (t-8ch) [inline src/main/java/org/sonar/plugins/stash/StashRequestFacade.java]: Hm, it should not matter. If they configured the environment and did not use it, the login should fail whether due to an exception by us or an invalid login to the stash server would not matter.

**Pull request merged**